Azure/azure-cli
# [ARM] Bump version of `azure-mgmt-managementgroups` from `0.2.0` to `1.0.0` (Track 2 SDK) and add new sub commands for management-group

**Description**


- Update the SDK version of azure-mgmt-managementgroups from 0.2.0 to 1.0.0.
- Add a new command subgroup `az account management-group entities`
- Add a new command subgroup `az account management-group hierarchy-settings`
- Add a new command subgroup `az account management-group tenant-backfill`
- Add a new command `az account management-group subscription show`
- Add a new command `az account management-group subscription show-sub-under-mg`
- Add a new command `az account management-group check-name-availability`

**Testing Guide**

- Check if "GroupName" is valid. 
  cmd: `az account management-group check-name-availability -n GroupName`

- Show the details of a subscription "subName" under a management group "GroupName"
  cmd: `az account management-group subscription show -n GroupName -s subName`

- Get the subscription(s) under a management group "GroupName"
  cmd: `az account management-group subscription show -n GroupName`

- Get the  Backfill Status for the current tenant
  cmd: `az account management-group tenant-backfill get`

- Start Backfilling Subscriptions for the current tenant
  cmd: `az account management-group tenant-backfill start`

- List all entities for the authenticated user
  cmd: `az account management-group entities list`

- Get all hierarchy settings under the "RootMG" tenant
  cmd: `az account management-group hierarchy-settings list-n RootMG`

- Create a hierarchy setting to require write permissions on the "RootMG" tenant to create a new Management Group under it
  cmd: `az account management-group hierarchy-settings create -n RootMG -r True`

- Create a hierarchy setting to set "defaultGroup" as the new MG groups get created under in the "RootMG" tenant
   cmd: `az account management-group hierarchy-settings create -n RootMG -m 
            /providers/Microsoft.Management/managementGroups/DefaultGroup`

- Update the hierarchy setting to require write permissions on the "RootMG" tenant to create a new Management Group under
  cmd: `az account management-group hierarchy-settings update-n RootMG -r True`

- Update the hierarchy setting to set "defaultGroup" as the new MG groups get created under in the "RootMG" tenant
   cmd: `az account management-group hierarchy-settings update -n RootMG -m 
             /providers/Microsoft.Management/managementGroups/DefaultGroup`

- Delete the hierarchy settings for the "RootMG" tenant
  cmd: `az account management-group hierarchy-settings delete -n RootMG`

**History Notes**

[ARM] `az account management-group entities`: Add a new command group to support entities (Management Groups and Subscriptions) operations for the authenticated user
[ARM] `az account management-group hierarchy-settings`: Add a new command group to support operations on hierarchy settings defined at the management group level
[ARM] `az account management-group tenant-backfill`: Add a new command group to support backfilling subscriptions for the tenant
[ARM] `az account management-group subscription show`: Get the details of a given subscription under a given management group
[ARM] `az account management-group subscription show-sub-under-mg`: Show what subscription is under a given management group
[ARM] `az account management-group check-name-availability`: Check if a management group name is valid and available

---

This checklist is used to make sure that common guidelines for a pull request are followed.
